core/src/test/java/com/linecorp/armeria/common/RequestContextTest.java

```diff
@@ -55,6 +55,25 @@
 
 class RequestContextTest {
 
+    @Test
+    void runWithRunnable() {
+        final RequestContext ctx = createContext();
+        ctx.run(() -> {
```

Comment:
How about add an `AtomicBoolean` and asserting it for checking the `Runnable` is sucessfully called.